### ISSUE TITLE
Incorrect IDE0004 (Cast is redundant) when negating enum value

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#18510]
**Version Used**: Visual Studio 2017 (15.1 - 26403.0)

**Steps to Reproduce**:

```csharp
enum Sign
{
    Positive = 1,
    Negative = -1
}

class T
{
    void Foo()
    {
        Sign mySign = Sign.Positive;
        Sign invertedSign = (Sign)(-(int)mySign);
    }
}
```

**Expected Behavior**:

IDE0004 is not reported for the `(int)` cast in `(int)mySign`.

**Actual Behavior**:

IDE0004 is reported for the `(int)` cast in `(int)mySign`. Removing the cast results in the following compilation error:

> CS0023: Operator '-' cannot be applied to operand of type 'Sign'

**Reference**:

This appears related to, but not quite a duplicate of #11008.

This was originally reported here:
https://developercommunity.visualstudio.com/content/problem/39596/intellisense-suggests-invalid-cast.html
